### ISSUE TITLE
HF/FSDP backend: pass `logits_to_keep` to avoid materializing full logits tensor

### ISSUE THREAD

**Issue by vsimkus** [NovaSky-AI/SkyRL#1858]
## Problem

The HF backend's [`model_wrapper.py`](https://vscode.dev/github/NovaSky-AI/SkyRL/blob/8cd5f165d81813c41d3e00b31391500b4a256d27/skyrl/backends/skyrl_train/workers/model_wrapper.py) computes logits for **all** sequence positions, then discards observation-position logits immediately after.
```python
action_log_probs = log_probs[:, -num_actions - 1 : -1]
```
The same is done for entropies in [`PolicyWorkerBase`.](https://github.com/NovaSky-AI/SkyRL/blob/8cd5f165d81813c41d3e00b31391500b4a256d27/skyrl/backends/skyrl_train/workers/worker.py) 

For models with large vocabularies (e.g., Qwen 3.5/3.6 with vocab size 248K), this materializes a massive tensor at the `lm_head`. At bf16, 262K tokens × 248K vocab = ~123 GiB.

Often, there are significantly fewer action tokens than observation tokens, so not materializing the observation token logits can produce significant savings.  

## Proposed fix

HuggingFace transformers models already support a [`logits_to_keep`](https://github.com/huggingface/transformers/blob/main/src/transformers/models/qwen3_5/modeling_qwen3_5.py#L1620-L1632) parameter that slices `hidden_states` **before** the `lm_head` projection:

```python
slice_indices = slice(-logits_to_keep, None) if isinstance(logits_to_keep, int) else logits_to_keep
logits = self.lm_head(hidden_states[:, slice_indices, :])
```

Since `num_actions` is already available in `forward()`, passing `logits_to_keep=num_actions + 1` would avoid computing logits for observation positions entirely. No custom kernels required.

**Standard path (no sample packing):** one-line change — pass `logits_to_keep=num_actions + 1` as an int.

**With `remove_microbatch_padding` / sample packing:** action positions are interleaved across packed sequences, so `logits_to_keep` needs the tensor form (indices of action positions).

## Context

The Megatron backend already solves this via fused linear cross-entropy (#1841, #1765). This issue is about the HF/FSDP backend, which has no equivalent optimization.

**Comment by NenoL2001** [comment]
Hi, I’d like to work on this issue.

My solution is:
pass logits_to_keep=num_actions + 1 for the standard non-packed HF/FSDP path;
handle packed batches using tensor indices when sequence_parallel_size == 1;
retain the current full-logits path as a safe fallback for sequence-parallel and unsupported model cases;
add numerical and gradient parity tests covering log probabilities and entropy;
report peak-memory results on a single H100 using a large-vocabulary Qwen configuration.

Please let me know if this scope sounds reasonable, or if you would prefer the standard non-packed path to be submitted first and packed-batch support handled in a follow-up PR.
